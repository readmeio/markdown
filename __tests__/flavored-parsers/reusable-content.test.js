import { mdast } from '../../index';

describe('reusable content transfomer', () => {
  it('parses camel cased content', () => {
    const md = '<TestComponent />';
    const tree = mdast(md);

    expect(tree.children[0].type).toBe('reusable-content');
    expect(tree.children[0].tag).toBe('TestComponent');
  });

  it('parses content in a container', () => {
    const md = '> <TestComponent />';
    const tree = mdast(md);

    expect(tree.children[0].type).toBe('blockquote');
    expect(tree.children[0].children[0].type).toBe('reusable-content');
    expect(tree.children[0].children[0].tag).toBe('TestComponent');
  });

  it('should replace a reusable content block if the block is provided', () => {
    const tags = {
      Test: `
# Test

[link](http://example.com)
    `,
    };
    const md = `
Before

<Test />

After
    `;

    const tree = mdast(md, { reusableContent: { tags } });

    expect(tree.children[0].children[0].value).toBe('Before');
    expect(tree.children[1]).toMatchSnapshot();
    expect(tree.children[2].children[0].value).toBe('After');
  });

  it('should replace a reusable content block with multiple words if the block is provided', () => {
    const tags = {
      MyCustomComponent: `
# Test

[link](http://example.com)
    `,
    };
    const md = '<MyCustomComponent />';

    const tree = mdast(md, { reusableContent: { tags } });

    expect(tree.children[0]).toMatchSnapshot();
  });

  it('should insert an empty node if the reusable content block is not defined', () => {
    const md = '<NotDefined />';
    const tree = mdast(md);

    expect(tree.children[0].type).toBe('reusable-content');
    expect(tree.children[0].children).toStrictEqual([]);
  });

  it('does not expand reusable content recursively', () => {
    const tags = {
      Test: '<Test />',
    };
    const md = '<Test />';
    const tree = mdast(md, { reusableContent: { tags } });

    expect(tree.children[0].children[0].type).toBe('reusable-content');
    expect(tree.children[0].children[0].children).toStrictEqual([]);
  });

  it('does not replace reusable content if it is disabled', () => {
    const tags = {
      Test: '<Test />',
    };
    const md = '<Test />';
    const tree = mdast(md, { reusableContent: { tags, disabled: true } });

    expect(tree.children[0].type).toBe('html');
    expect(tree.children[0].value).toBe('<Test />');
  });
});
