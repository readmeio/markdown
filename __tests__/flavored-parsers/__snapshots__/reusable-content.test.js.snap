// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reusable content transfomer should replace a reusable content block if the block is provided 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 7,
              "line": 2,
              "offset": 7,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 2,
              "offset": 3,
            },
          },
          "type": "text",
          "value": "Test",
        },
      ],
      "data": Object {
        "hProperties": Object {
          "id": "test",
        },
        "id": "test",
      },
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 7,
          "line": 2,
          "offset": 7,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 6,
                  "line": 4,
                  "offset": 14,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 4,
                  "offset": 10,
                },
              },
              "type": "text",
              "value": "link",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 27,
              "line": 4,
              "offset": 35,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 4,
              "offset": 9,
            },
          },
          "title": null,
          "type": "link",
          "url": "http://example.com",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 27,
          "line": 4,
          "offset": 35,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 9,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Position {
    "end": Object {
      "column": 1,
      "line": 6,
      "offset": 19,
    },
    "indent": Array [
      1,
      1,
    ],
    "start": Object {
      "column": 1,
      "line": 4,
      "offset": 9,
    },
  },
  "tag": "Test",
  "type": "reusable-content",
}
`;

exports[`reusable content transfomer should replace a reusable content block with multiple words if the block is provided 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "position": Position {
            "end": Object {
              "column": 7,
              "line": 2,
              "offset": 7,
            },
            "indent": Array [],
            "start": Object {
              "column": 3,
              "line": 2,
              "offset": 3,
            },
          },
          "type": "text",
          "value": "Test",
        },
      ],
      "data": Object {
        "hProperties": Object {
          "id": "test",
        },
        "id": "test",
      },
      "depth": 1,
      "position": Position {
        "end": Object {
          "column": 7,
          "line": 2,
          "offset": 7,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "heading",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "position": Position {
                "end": Object {
                  "column": 6,
                  "line": 4,
                  "offset": 14,
                },
                "indent": Array [],
                "start": Object {
                  "column": 2,
                  "line": 4,
                  "offset": 10,
                },
              },
              "type": "text",
              "value": "link",
            },
          ],
          "position": Position {
            "end": Object {
              "column": 27,
              "line": 4,
              "offset": 35,
            },
            "indent": Array [],
            "start": Object {
              "column": 1,
              "line": 4,
              "offset": 9,
            },
          },
          "title": null,
          "type": "link",
          "url": "http://example.com",
        },
      ],
      "position": Position {
        "end": Object {
          "column": 27,
          "line": 4,
          "offset": 35,
        },
        "indent": Array [],
        "start": Object {
          "column": 1,
          "line": 4,
          "offset": 9,
        },
      },
      "type": "paragraph",
    },
  ],
  "position": Position {
    "end": Object {
      "column": 2,
      "line": 3,
      "offset": 24,
    },
    "indent": Array [
      1,
      1,
    ],
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "tag": "MyCustomComponent",
  "type": "reusable-content",
}
`;
